content-src/components/Search/Search.js

```diff
@@ -1,27 +1,318 @@
+      <button ref="performSearchButton"
+        onClick={e => { e.preventDefault(); this.performSearch({engineName: currentEngine.name, searchString});}}>
         <span className="sr-only">Search</span>
       </button>
+      <div className="search-container" role="presentation" hidden={!this.getDropdownVisible()}>
+      <section className="search-title" hidden={!formHistory.concat(suggestions).length}>
+        <img id="current-engine-icon" src={currentEngine.iconBuffer} alt={currentEngine.name.charAt(0)} width="16px" height="16px"/>
+        {this.props.searchHeader.replace("%S", currentEngine.name)}
+      </section>
+      <section className="history-search-suggestions" hidden={!formHistory.length}>
+        <ul ref="formHistoryList" role="listbox">
+          {formHistory.map(suggestion => {
+            const active = (this.state.activeSuggestionIndex === suggestionsIdIndex);
+            const activeEngine = this.getActiveEngine();
+            return (<li key={suggestion} role="presentation">
+                  <a id={"history-search-suggestions-" + suggestionsIdIndex++ }
+                     className={active ? "active" : ""} role="option"
+                     aria-selected={active} onClick={() => this.performSearch({engineName: activeEngine.name, searchString: suggestion})}>
+                     <div id="historyIcon" className={active ? "active" : ""}></div>{suggestion}</a>
+                </li>);
+          })}
+        </ul>
+      </section>
+      <section className="search-suggestions" hidden={!suggestions.length}>
+        <ul ref="suggestionsList" role="listbox">
+          {suggestions.map(suggestion => {
+            const active = (this.state.activeSuggestionIndex === suggestionsIdIndex);
+            const activeEngine = this.getActiveEngine();
+            return (<li key={suggestion} role="presentation">
+            <a id={"search-suggestions-" + suggestionsIdIndex++ }
+              className={active ? "active" : ""} role="option"
+              aria-selected={active}
+              onClick={() => this.performSearch({engineName: activeEngine.name, searchString: suggestion})}>{suggestion}</a>
+            </li>);
+          })}
+        </ul>
+      </section>
+      <section className="search-title">
+           <span>{this.props.searchForSomethingWith.replace("%S", searchString)}</span>
+      </section>
+      <section className="search-partners" role="group">
+            <ul>
+            {engines.map(option => {
+              const icon = option.icon;
+              const active = (this.state.activeEngineIndex === enginesIdIndex);
+              return (<li key={option.name} className={active ? "active" : ""}>
+                <a id={"search-partners-" + enginesIdIndex++ } aria-selected={active}
+                      onClick={() => this.performSearch({engineName: option.name, searchString: this.getActiveSuggestion()})}>
+                <img src={icon} alt={option.name} width="16px" height="16px"/></a>
```

Comment:
I don't think you need to specify `"px"` units here when using the `width` and `height` attributes (only in CSS).